src/core/Akka/Actor/ActorPath.cs

```diff
@@ -325,42 +412,58 @@ public static ActorPath Parse(string path)
         /// <returns>TBD</returns>
         public static bool TryParse(string path, out ActorPath actorPath)
         {
-            actorPath = null;
+            if (!TryParseAddress(path, out var address, out var absoluteUri))
+            {
+                actorPath = null;
+                return false;
+            }
 
-            if (!TryParseAddress(path, out var address, out var absoluteUri)) return false;
-            var spanified = absoluteUri;
+            return TryParse(new RootActorPath(address), absoluteUri, out actorPath);
+        }
 
-            // check for Uri fragment here
-            var nextSlash = 0;
+        /// <summary>
+        /// Tries to parse the uri, which should be a uri not containing protocol.
+        /// For example "/user/my-actor"
+        /// </summary>
+        /// <param name="basePath">the base path, normaly a root path</param>
+        /// <param name="absoluteUri">TBD</param>
+        /// <param name="actorPath">TBD</param>
+        /// <returns>TBD</returns>
+        public static bool TryParse(ActorPath basePath, ReadOnlySpan<char> absoluteUri, out ActorPath actorPath)
+        {
+            actorPath = basePath;
```

Comment:
interesting - so we only parse the new parts of each successive path element. That's why this is so much faster in the benchmarks. Makes sense.

Reply:
original always parsed the RootActorPath including the address.
We now use it to rebase paths and we cache the root to deduplicate it.